todoapp/src/App.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import Todo from './Todo';
+import todos from './todo.json';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+     this.state = {
+      todos: [],
+      newToDoItem: {
+      text: ''        
+      }
+    };
+
+    this.changeToDoList = this.changeToDoList.bind(this);
+    this.inputAdd = this.inputAdd.bind(this);
+    this.submitNewItem = this.submitNewItem.bind(this);
+  }
+  componentDidMount() {
+      setTimeout(() => {
+      this.setState({ todos });
+    }, 1000);
+  }
+
+  changeToDoList(updatedReadValue, toDoId) {
+    const updatedtodos = this.state.todos.map(comment => {
+      if (toDoId === comment.id) {
+        return Object.assign({}, comment, {read: updatedReadValue});
+      }
+      return comment;
+    });
+    this.setState({
+      todos: updatedtodos
+    });
+  }
+
+  rendertodos(todos) {
+    // Conditional rendering
+    if (todos.length === 0) {
+      return <div>Loading...</div>;
+    }
+    return todos.map(comment => {
+      return (
+        <Todo
+          key={comment.id}
+          comment={comment}
+          onReadChange={this.changeToDoList}
+        />
+      );
+    })
+  }
+
+  getNumberOfReadtodos(todos) {
+    return todos.filter(comment => !comment.read).length;
+  }
+
+  submitNewItem(event) {
+    event.preventDefault();
+    const updatednewToDoItem = Object.assign({}, this.state.newToDoItem, {
```

Comment:
Like the previous comment about `Object.assign`
```suggestion
    const updatednewToDoItem = {...this.state.newToDoItem, 
      date: new Date(),
      id: this.state.todos.length + 1,
      read: false
    }
```